cns/requestcontroller/kubecontroller/crdrequestcontroller.go

```diff
@@ -239,40 +241,52 @@ func (crdRC *crdRequestController) initCNS() error {
 	}
 
 	// Convert to CreateNetworkContainerRequest
-	if ncRequest, err = CRDStatusToNCRequest(nodeNetConfig.Status); err != nil {
+	ncRequest, err := crdStatusToNCRequest(nodeNetConfig.Status)
+	if err != nil {
 		logger.Errorf("Error when converting nodeNetConfig status into CreateNetworkContainerRequest: %v", err)
 		return err
 	}
 
-	// Get all pods using direct client
-	if pods, err = crdRC.getAllPods(cntxt, crdRC.nodeName); err != nil {
-		logger.Errorf("Error when getting all pods when initializing cns: %v", err)
-		return err
-	}
+	var podInfoByIPProvider cns.PodInfoByIPProvider
 
-	// Convert pod list to map of pod ip -> kubernetes pod info
-	if len(pods.Items) != 0 {
-		podInfoByIP = make(map[string]cns.KubernetesPodInfo)
-		for _, pod = range pods.Items {
-			//Only add pods that aren't on the host network
-			if !pod.Spec.HostNetwork {
-				podInfo = cns.KubernetesPodInfo{
-					PodName:      pod.Name,
-					PodNamespace: pod.Namespace,
-				}
-				podInfoByIP[pod.Status.PodIP] = podInfo
-			}
+	if crdRC.cfg.InitializeFromCNI {
```

Comment:
I dont think this alone is definining how to reconcile. Rather you need to check GlobalPodInfoVersion